nemo/collections/common/parts/adapter_modules.py

```diff
@@ -61,6 +61,24 @@ def get_default_strategy_config(self) -> 'dataclass':
         """
         return adapter_mixin_strategies.ResidualAddAdapterStrategyConfig()
 
+    def adapter_unfreeze(self,):
+        """
+        Sets the requires grad for all parameters in the adapter to True.
+        This method should be overridden for any custom unfreeze behavior that is required.
+        For example, if not all params of the adapter should be unfrozen.
+        """
+        for param in self.parameters():
+            param.requires_grad_(True)
+        self.train()
+    
+    def adapter_freeze(self,):
```

Comment:
@titu1994 added adapter_freeze and adapter_unfreeze hooks in the adapter mixin per your suggestion. `adapter_unfreeze` is called internally when `unfreeze_enabled_adapters` is called. but there is no equivalent call for `freeze_enabeld_adapters` in the mixin -- So currently `adapter_freeze` will not be triggered at all.

Reply:
Its ok, its for mirrored api of freeze and unfreeze of adapter. For freeze_enabled_adapters, you can do model.unfreeze() so that's ok.